README.md

```diff
@@ -3,4 +3,135 @@
 
 # MQTT bridge for Apache Kafka®
 
-This project provides a software component which acts as a bridge between [MQTT 3.1.1](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html) and an [Apache Kafka®](https://kafka.apache.org/) cluster.
+This project provides a software component which acts as a bridge between [MQTT 3.1.1](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html) 
+and an [Apache Kafka®](https://kafka.apache.org/) cluster.
+
+## Running the bridge
+
+### Locally
+
+Clone the repository and update your dependencies using maven.
+Afterward, edit the `config/application.properties` file which contains the configuration. And provide a valid [topic mapping rules file]().
+
+Once your configuration is ready, start the bridge using:
+
+```bash
+<application> --config-file config/application.properties --mapping-rules=path/to/mapping-rules
+```
+
+You can find more information about the topic mapping rules in the MQTT Bridge.
+
+### Running Tests
+
+```bash
+mvn clean test
+```
+
+## MQTT bridge documentation
+
+This section provides information about the MQTT Bridge. It describes the MQTT Bridge, how it works, and how to configure it.
+
+### 1. MQTT Bridge Overview
+
+#### 1.1 How it works
+
+To enable a seamless integration between MQTT and Kafka, the MQTT Bridge provides a way to map MQTT topics to Kafka topics. 
+This mapping is done using a set of predefined patterns and kafka topic templates. The MQTT Bridge uses these patterns to map MQTT topics to Kafka topics. 
+As a part of the bridge, a Kafka producer will be responsible for producing messages from the MQTT clients to the Kafka Cluster.
+
+#### 1.2. Topic Mapping Rules(TOMAR)
+
+The TOMAR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic names.
+A Mapping Rule is a model that contains an `MQTT topic pattern` and a `Kafka topic template`. 
+All the incoming MQTT message's topic should match an `MQTT topic pattern` in the `TOMAR` so that the bridge knows in which 
```

Comment:
```suggestion
All the incoming MQTT message's topic should match an `MQTT topic pattern` in the `TOMAR` so that the bridge knows in which Kafka topic to produce this message. 
```